src/nvim/os/input.c

```diff
@@ -7,13 +7,18 @@
 
 #include <uv.h>
 
+// This needs to be included before we define EXTERN
+#include "nvim/ui.h"
+#include "nvim/msgpack_rpc/channel.h"
+
+#undef EXTERN
+#define EXTERN
```

Comment:
what is this for?

Reply:
`EXTERN` gets defined by macros.h to:

``
7 # define EXTERN extern
``

which gets included by globals.h. globals.h gets included by ui.h. But this fine wants to define global variables with EXTERN defined to extern. So we need to undefine the EXTERN from macros.h ...

Reply:
s/fine/file/

Reply:
I've added comments to make it clearer in the code.